src/main/capi/data_chunk-c.cpp

```diff
@@ -174,6 +174,38 @@ duckdb_vector duckdb_array_vector_get_child(duckdb_vector vector) {
 	return reinterpret_cast<duckdb_vector>(&duckdb::ArrayVector::GetEntry(*v));
 }
 
+duckdb_vector duckdb_map_vector_get_keys(duckdb_vector vector) {
+	if (!vector) {
+		return nullptr;
+	}
+	auto v = reinterpret_cast<duckdb::Vector *>(vector);
+	return reinterpret_cast<duckdb_vector>(&duckdb::MapVector::GetKeys(*v));
+}
+
+duckdb_vector duckdb_map_vector_get_values(duckdb_vector vector) {
+	if (!vector) {
+		return nullptr;
+	}
+	auto v = reinterpret_cast<duckdb::Vector *>(vector);
+	return reinterpret_cast<duckdb_vector>(&duckdb::MapVector::GetValues(*v));
+}
+
+duckdb_vector duckdb_union_vector_get_tags(duckdb_vector vector) {
+	if (!vector) {
+		return nullptr;
+	}
+	auto v = reinterpret_cast<duckdb::Vector *>(vector);
+	return reinterpret_cast<duckdb_vector>(&duckdb::UnionVector::GetTags(*v));
+}
+
+duckdb_vector duckdb_union_vector_get_member(duckdb_vector vector, idx_t tag) {
+	if (!vector) {
+		return nullptr;
+	}
+	auto v = reinterpret_cast<duckdb::Vector *>(vector);
+	return reinterpret_cast<duckdb_vector>(&duckdb::UnionVector::GetMember(*v, tag));
```

Comment:
`tag` has to be verified first, otherwise we'll hit:
```c++
D_ASSERT(member_index < UnionType::GetMemberCount(vector.GetType()));
```

Which would result in an InternalException instead